.circleci/config.yml

```diff
@@ -178,18 +178,39 @@ jobs:
       - store_test_output
       - fail_fast
       - teardown_vm
-  'Unit Tests':
+  'Local Unit Tests':
     executor:
       name: amphtml-large-executor
     steps:
+      - unless:
+          condition:
+            matches:
+              pattern: ^pull/\d+$
+              value: << pipeline.git.branch >>
+          steps:
+            - run:
+                name: 'Skip Local Unit Test on Push Builds'
+                command: circleci-agent step halt
```

Comment:
The format here is a bit different - this is a _step_, not a _job_ that needs to be conditioned. I tried setting the job to skip entirely, but then the dependent job (`All Unit Tests`) never executes. As a middle-group, I moved this block to the `jobs:` section - does this look good for you?

Reply:
To clarify, I was suggesting that the condition implementation be added at the top as follows:

```yaml
pr_builds_only: &pr_builds_only
  filters:
    branches:
      ignore:
        - main
        - /^amp-release-.*$/
```

Then, it can be used by a step as follows:

```yaml
  command_name:
    steps:
      <<: *pr_builds_only
      - run:
          command: <command to run during pr builds>
```

This way, it can also be reused in future by other jobs / steps. (And if this particular step needs to run for just push builds, you could similarly reuse the `push_builds_only` condition.)

Reply:
This isn't a `filter` though - a `filter` controls whether an entire _job_ runs or not, while this particular case requires that one individual _step_ be run - I don't want to create `pr_builds_only` defined as a step-controller, because `push_builds_only` is a job-controller - so there's a subtle inconsistency here that could hurt us down the line

Reply:
Your point about filters has convinced me. Okay to leave this as is 👍